src/modules/auth/token.service.ts

```diff
@@ -64,4 +69,28 @@ export class TokenService {
   async invalidateRefreshToken(id: string) {
     return this.refreshTokenRepository.deleteRefreshTokenById(id);
   }
+
+  async handleOriginCheck(req: Request, res: Response, next: NextFunction) {
+    let token = null;
+    const requestOrigin = req.headers['origin'];
+
+    if (req.headers['authorization']) {
+      token = req.headers['authorization'].replace('Bearer ', '');
+    } else {
+      token = req.cookies.token
+    }
+
+    if (token) {
+      const decodedToken = this.jwtService.decode(token) as TokenPayload;
+      if (decodedToken.supportedOrigins.includes(requestOrigin) ||
+        decodedToken.supportedHosts.includes(req.hostname) // swagger and server based requests check only
```

Comment:
the second check is written only for swagger and server-based requests

as their requests are from the same domain as the servers domain, **origin** header is missing in requests, so I am checking if the requested URL is the same as servers HOST(like server is requesting to server), then check should pass.

Reply:
I don't really understand how the `HOST` header is relevant here. Isn't the `HOST` the destination of the request? So I don't understand how it says anything about the client.